CDCgov/prime-reportstream
# 13507: FHIR Converter integration tests

This PR applies a fix for which message model is chosen when parsing an HL7 message by deferring to the topic as well as adding comprehensive integration tests for the FHIR converter

Test Steps:
1. Send one of the elims messages through the pipeline and verify that it gets delivered
2. Send one of the MarsOtc messages use to test validation through the pipeline and verify it gets delivered

Verify the coverage (generated from only running the new tests)
<img width="1118" alt="Screenshot 2024-04-26 at 2 49 48 PM" src="https://github.com/CDCgov/prime-reportstream/assets/1225895/af2328c1-20cb-4675-9070-327e34d00aba">

## Changes
- Adds the ability to use an HL7MessageParseConfiguration set on the topic (required to support the RadXMars spec), otherwise uses the current fallback behavior of 
  - Any ORU_R01 message is parsed as 27 and otherwise the default HAPI behavior is use
- Adds a comprehensive set of integration tests for the FHIR Converter

## Checklist

### Testing

### Process

## Linked Issues
- #13507 

## To Be Done


## Specific Security-related subjects a reviewer should pay specific attention to
